### PR TITLE
add support for packages of type roundcube-skin, fix sql execution an…

### PR DESCRIPTION
**Pull request by johndoh**
…d other improvements

update of #24 for Composer V2.

I thought it might be handy to extend the plugin-installer to support packages of type roundcube-skin and not just roundcube-plugin that way third party skins can be distributed in the same way as plugins through composer.

Add some bug fixes:

    #15 check for a config file on .inc.php in the roundcube config dir before creating confic.inc.php in the plugin dir
    #23 small fix to avoid null reference error
    #19, #21 replace use of rcubeinitdb.sh and updatedb.sh with rcmail_utils::db_init() and rcmail_utils::db_update()
    if the config var was not already in config.inc.php then it was added with a extra semi colon at the end eg $config['plugins'] = array('plugin_name');; this has been fixed.

Update: ref 3701ac864991006efdee388b565831c25389f1ab. I have added a more flexible fix for this issue. `config.inc.php` is assumed to be the only file that needs to be saved but incase there are others or its in a differnt location plugin devs can specify `persistent-files` in their compose.json file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,29 +13,56 @@ This installer ensures that plugins end up in the correct directory:
...
+    {
+        "name": "<your-vendor-name>/<skin-name>",
+        "type": "roundcube-skin",
+        "license": "GPL-3.0+",
+        "require": {
+            "roundcube/plugin-installer": ">=0.3.0"
+        }
+    }
+
+## Roundcube specifc composer.json params
```
Comment (alecpl) [inline README.md:35]: a typo in "specifc".

File: src/Roundcube/Composer/PluginInstaller.php [hunk 2]
```diff
@@ -2,347 +2,60 @@
...
-     */
     public function getVendorDir()
     {
-        $pluginDir  = getcwd();
-        $pluginDir .= '/plugins';
+        $package_dir  = getcwd();
+        $package_dir .= DIRECTORY_SEPARATOR . 'plugins';
 
-        return $pluginDir;
+        return $package_dir;
```
Comment (alecpl) [inline src/Roundcube/Composer/PluginInstaller.php:22]: This method could be a one-liner.

**Review by alecpl — CHANGES_REQUESTED** [review]

**Review by alecpl — APPROVED** [review]
Approved, but @thomascube should also review this.

**Comment by thomascube** [comment]
@johndoh Do you have an example skin that can be installed with this installer? I just like to see it work :-)

**Comment by johndoh** [comment]
sure. if you add this to your `composer.json`:
```
    "repositories": [
        ...
        {
            "type": "vcs",
            "url": "https://github.com/johndoh/plugin-installer"
        },
        {
            "type": "vcs",
            "url": "https://github.com/johndoh/roundcube-sample-skin"
        }
    ],
    ...
    "require": {
        ...
        "roundcube/plugin-installer": "dev-roundcube-skin",
        "johndoh/sample-skin": "dev-master"
    }
```
it will install a very simple skin which extends Elastic. It changes the background colour of the watermark template (to blue) and changes the icons for the markasjunk plugin toolbar items (to thumbs up/down)

**Pull request merged**